cluster-autoscaler/cloudprovider/packet/packet_price_model.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package packet
+
+import (
+	"math"
+	"time"
+
+	apiv1 "k8s.io/api/core/v1"
+)
+
+// PacketPriceModel implements PriceModel interface for Packet.
+type PacketPriceModel struct {
+}
+
+var instancePrices = map[string]float64{
+	"c1.large.arm":  0.5000,
+	"c1.small.x86":  0.4000,
+	"c1.xlarge.x86": 1.7500,
+	"c2.large.arm":  1.0000,
+	"c2.medium.x86": 1.0000,
+	"c3.medium.x86": 1.1000,
+	"c3.small.x86":  0.5000,
+	"g2.large.x86":  5.0000,
+	"m1.xlarge.x86": 1.7000,
+	"m2.xlarge.x86": 2.0000,
+	"n2.xlarge.x86": 2.2500,
+	"s1.large.x86":  1.5000,
+	"s3.xlarge.x86": 1.8500,
+	"t1.small.x86":  0.0700,
+	"t3.small.x86":  0.3500,
+	"x1.small.x86":  0.4000,
+	"x2.xlarge.x86": 2.5000,
+}
+
+// NodePrice returns a price of running the given node for a given period of time.
+// All prices are in USD.
+func (model *PacketPriceModel) NodePrice(node *apiv1.Node, startTime time.Time, endTime time.Time) (float64, error) {
+	price := 0.0
+	if node.Labels != nil {
+		if machineType, found := node.Labels[apiv1.LabelInstanceType]; found {
+			if pricePerHour, found := instancePrices[machineType]; found {
+				price = pricePerHour * getHours(startTime, endTime)
+			}
+		}
+	}
+	return price, nil
+}
+
+func getHours(startTime time.Time, endTime time.Time) float64 {
+	minutes := math.Ceil(float64(endTime.Sub(startTime)) / float64(time.Minute))
+	hours := minutes / 60.0
+	return hours
+}
+
+// PodPrice returns a theoretical minimum price of running a pod for a given
+// period of time on a perfectly matching machine.
+func (model *PacketPriceModel) PodPrice(pod *apiv1.Pod, startTime time.Time, endTime time.Time) (float64, error) {
+	// Always return 1.0 since Packet is only charging per Node.
+	return 1.0, nil
```

Comment:
The is not about charging for pod. The price expander uses a ration between theoretical cost of resources used by a pod and the actual cost of infrastructure. 
This method is supposed to estimate cost of cpu, memory, etc used by the pod. In other words how much a custom-shaped VM that would have exactly the resources (cpu, memory, gpu, ...) used by the pod would cost if one was available.

Reply:
Thank you for your explanation, I updated this function. As far as I understand, this function is used in the expander's heuristic. Since, there is no native support for custom VMs, or a pricing scheme that is per CPU/RAM usage in Packet (it is a bare metal provider after all), I calculated the constants (`cpuPricePerHour`, `memoryPricePerHourPerGb`) as the theoretical best value you can get per hour from the available instances. If you think that there is a better approach to this, please let me know.

Reply:
I believe @mwielgus did the same for GCE - there are custom VMs there, but the pricing is not the same. It doesn't need to be perfect it's just used when trying to estimate how effective the packing of pods is wrt cost of underlying resources.

Reply:
Ok, I think I have done the same thing for Packet with the only difference being that I follow Packet's pricing model. Do you think that there is any other issue that needs to be addressed?